### ISSUE TITLE
showing false error for ~/.rd/bin directory not being present in PATH variable

### ISSUE THREAD

**Issue by humancalico** [rancher-sandbox/rancher-desktop#3673]
### Actual Behavior

Rancher Desktop is not able to detect the PATH variable entry `~/.rd/bin`

### Steps to Reproduce

1. Start with a fresh configuration for your shell
2. Add `~/.rd/bin` to your PATH variable by editing your `~/.profile` file and adding
	```sh
	export PATH=$HOME/.rd/bin:$PATH
	```
3. Load your `~/.profile` file by zsh login shell by adding this line to your `~/.zprofile`
	```zsh
	[[ -e ~/.profile ]] && emulate sh -c 'source ~/.profile'
	```
4. Reload your shell
5. ```sh
	echo $PATH
	# /Users/akshat/.rd/bin:/usr/bin:/bin
	```

### Result

```
The ~/.rd/bin directory has not been added to the PATH, so command-line utilities are not configured in your zsh shell.
```

Seeing the above error on the diagnostics panel in Rancher Desktop.

### Expected Behavior

It shouldn't have given this error in the diagnostics panel.

### Additional Information

_No response_

### Rancher Desktop Version

1.7.0

### Rancher Desktop K8s Version

1.23.4

### Which container engine are you using?

containerd (nerdctl)

### What operating system are you using?

macOS

### Operating System / Build Version

macOS Ventura 13.1

### What CPU architecture are you using?

arm64 (Apple Silicon)

### Linux only: what package format did you use to install Rancher Desktop?

None

### Windows User Only

_No response_

**Comment by mook-as** [comment]
Thanks for the bug report!

We run our [diagnostics](https://github.com/rancher-sandbox/rancher-desktop/blob/main/pkg/rancher-desktop/main/diagnostics/rdBinInShell.ts) using `zsh -i -c 'echo "$PATH"'` (effectively); could you check if the produces the expected output? (You'll have to run it from a shell that _doesn't_ have `~/.rd/bin` in the `PATH`, of course.)  It's possible that the arguments we're passing to `zsh` are incorrect here.

**Comment by humancalico** [comment]
```zsh
zsh -i -c 'echo "$PATH"'  
/Users/akshat/.rd/bin:/usr/local/bin:/usr/bin:/bin
```
Have not explicitly added `~/.rd/bin` to zsh. Only to `~/.profile` and have linked `~/.profile` with `~/.zprofile` by adding
```
[[ -e ~/.profile ]] && emulate sh -c 'source ~/.profile'
```

**Comment by JDTX0** [comment]
@mook-as Ran into this as well, after some debugging I found the cause.

TLDR; ZSH doesn't execute the contents of `~/.zprofile` for non-login shells. (`-l` flag)

As an example, I added `export THIS_SHOULD_RUN_FROM_ZSH_RC=1` to `~/.zshrc` and `export THIS_SHOULD_RUN_FROM_ZPROFILE=1` to `~/.zprofile`.

Then I ran `zsh` with `-x` to see when/where it was getting set:

Interactive shell only:
```
  zsh -x -i -c 'echo $RAN' 2>&1 | grep "THIS_SHOULD_RUN"
+/Users/jdt/.zshrc:36> export THIS_SHOULD_RUN_FROM_ZSH_RC=1
```

Notice that the contents of `~/.zprofile` aren't executed at all. Switching the `-i` flag to `-l` (login shell) makes `~/.zprofile` get executed:
```
zsh -x -l -c 'echo $RAN' 2>&1 | grep "THIS_SHOULD_RUN"
+/Users/jdt/.zprofile:1> export THIS_SHOULD_RUN_FROM_ZPROFILE=1
```

Passing both `-i` and `-l` makes both zshrc + zprofile execute: 
```
zsh -x -i -l -c 'echo $RAN' 2>&1 | grep "THIS_SHOULD_RUN"
+/Users/jdt/.zprofile:1> export THIS_SHOULD_RUN_FROM_ZPROFILE=1
+/Users/jdt/.zshrc:36> export THIS_SHOULD_RUN_FROM_ZSH_RC=1
```

The reason the .rd bin appears in the path in a standard shell is because a standard shell is both an interactive AND login shell. The diagnostics code only spawns an interactive shell, not login, so it doesn't end up executing anything set in zprofile. 

Here's some documentation on that: https://zsh.sourceforge.io/Guide/zshguide02.html
```
~/.zprofile
Run for login shells.
[...]
~/.zshrc
Run for interactive shells.
```

I see two possible solutions:
1) Add the `-l` flag (login shell) to the ZSH arguments. This has the downside of possibly slowing diagnostics down, as all code within `~/.zprofile` would now get executed when it wasn't previously.
2) Update the error message to explicitly state that the RD `$PATH` is missing from `~/.zshrc` so as to point the user to the correct file to fix

I would lean towards solution #1, just because there may be a lot of others that may expect the zprofile approach to work.

**Comment by jandubois** [comment]
> TLDR; ZSH doesn't execute the contents of `~/.zprofile` for non-login shells. (`-l` flag)

Yes, but Rancher Desktop only adds the `PATH` modification code to `~/.zshrc` and not to `~/.zprofile`: https://github.com/rancher-sandbox/rancher-desktop/blob/7abe4c3bbba0c4f453afe10548c421bf9e876956/pkg/rancher-desktop/integrations/pathManager.ts#L121

You can see in https://github.com/rancher-sandbox/rancher-desktop/issues/2951#issuecomment-1254170775 that `zsh -l -c` does **not** produce the right result (precisely because we don't modify `~/.zprofile`; maybe we should?).

I would like to understand how Rancher Desktop seems to modify `.zprofile` on your system instead of `.zshrc`; it doesn't seem to be possible from the source code.

Or did you select "Manual" `PATH` configuration, and have added the change just to `.zprofile` manually? I guess this possibility alone is a good argument for running the test via `zsh -i -l -c`.

**Comment by jandubois** [comment]
> I guess this possibility alone is a good argument for running the test via `zsh -i -l -c`.

I think it is, so I've created #3704 to add the `-l` flag. This should address the scenario described above by @humancalico.

**Comment by JDTX0** [comment]
> Or did you select "Manual" `PATH` configuration, and have added the change just to `.zprofile` manually? I guess this possibility alone is a good argument for running the test via `zsh -i -l -c`.

Exactly, did a manual edit. I rarely trust anything automatically modifying my zshrc/zprofile as some things I use (fig for instance) require specific ordering, but that's another topic all together.
